ci/azure-windows-36.yaml

```diff
@@ -6,18 +6,16 @@ dependencies:
   - blosc
   - bottleneck
   - fastparquet
-  - feather-format
   - matplotlib
   - numexpr
   - numpy=1.14*
   - openpyxl=2.5.5
-  - pyarrow
+  - pyarrow=0.9.0
```

Comment:
I could reproduce #23180 and the first fix I tried (`boost-cpp<1.67`) worked. Pinning `pyarrow` is probably not gonna change things, because the offending error happens with `0.9.0` anyway.

Reply:
https://repo.continuum.io/pkgs/main/win-64/ has a package for pyarrow 0.9.0. I'm hoping that removing feather-format will keep us in main, not not conda-forge for these packages.

Reply:
But, if this fails I'll try pinning to boost-cpp<1.67

Reply:
It failed. Trying a pin now.